### PR TITLE
feat: conditional formatting show applied rules on hover

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/4197

### Description:

- [x] on hover, I can see the rules applied to a cell that has conditional formatting applied to it (i.e. only cells with colour show anything on hover. unaffected cells don't show anything on hover.)
- [x] Here are some examples of rules and what would show in the hover:
Rule 1:
  - product_value is greater than 5 (and condition 2:) is less than 10
  - on hover, it would show: is greater than 5 and is less than 10

CleanShot 2023-01-18 at 20 10 02@2x

### CODE REVIEW

File: packages/frontend/src/components/common/Filters/configs.ts [hunk 1]
```diff
@@ -148,26 +150,36 @@ export const getFilterRuleLabel = (
...
-                    filterRule.settings.unitOfTime.length - 1,
-                )}`;
+                if (isFilterRule(filterRule)) {
+                    valuesText = `${filterRule.settings.unitOfTime.substring(
+                        0,
+                        filterRule.settings.unitOfTime.length - 1,
+                    )}`;
+                } else {
+                    valuesText = `${filterRule.values?.[0]}`;
+                }
```
Comment (IrakliJani) [inline packages/frontend/src/components/common/Filters/configs.ts:170]: I am reusing filter methods because conditional formatting uses components from the filters. and they are basically the same. I created a "supertype" called ConditionalRule which is a supertype for `ConditionalFormattingRule` and `FilterRule` (they both extend from it). I am not sure if it is a good idea to reuse this `getFilterRuleLabel` (now named as `getConditionalRuleLabel`) but again, it is very same. except the date difference which conditional formatting does not support (yet?)

File: packages/frontend/src/components/common/Filters/configs.ts [hunk 2]
```diff
@@ -192,7 +204,7 @@ export const getFilterRuleLabel = (
 };
 
 export const getFilterRuleTables = (
-    filterRule: FilterRule | DashboardFilterRule,
+    filterRule: ConditionalRule,
```
Comment (IrakliJani) [inline packages/frontend/src/components/common/Filters/configs.ts:207]: `DashboardFilterRule` extends `FilterRule` and `FilterRule` extends `ConditionalRule`

File: packages/frontend/src/components/common/Filters/configs.ts [hunk 3]
```diff
@@ -122,8 +124,8 @@ type FilterRuleLabels = {
     value?: string;
 };
 
-export const getFilterRuleLabel = (
-    filterRule: FilterRule,
+export const getConditionalRuleLabel = (
```
Comment (owlas) [inline packages/frontend/src/components/common/Filters/configs.ts:127]: do we have tests for these labels? I'm finding it hard to visualise how the final labels will look in each use case!

**Review by owlas — COMMENTED** [review]
Two questions otherwise looks good!

File: packages/common/src/types/filter.ts [hunk 4]
```diff
@@ -67,7 +67,7 @@ export type DashboardFilterRule<
 };
 
 export const isDashboardFilterRule = (
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/common/src/types/filter.ts:69]: @owlas 

short answer - yes.

tldr;
so basically, `ConditionalRule` is the base of all, contains only a rule and values which is needed for conditional formatting. we can call this a `Rule` maybe? I'm open to rename and make it more obvious if it's not

```ts
export type ConditionalRule<O = ConditionalOperator, V = unknown> = {
    operator: O;
    values?: V[];
};

// FilterRule extends ConditionalRule (adds id and settings object)
export interface FilterRule<
    O = FilterOperator,
    T = FieldTarget,
    V = any,
    S = any,
> extends ConditionalRule<O, V> {
    id: string;
    target: T;
    settings?: S;
}

// DashboardFilterRule extends FilterRule and adds tileTargets record and label
export type DashboardFilterRule<
    O = FilterOperator,
    T = DashboardFieldTarget,
    V = any,
    S = any,
> = FilterRule<O, T, V, S> & {
    tileTargets?: Record<string, DashboardFieldTarget>;
    label: undefined | string;
};
```

File: packages/frontend/src/components/common/Filters/configs.ts (hunk 3 above)
Comment (IrakliJani) [reply packages/frontend/src/components/common/Filters/configs.ts:127]: @owlas no we don't. do you think it is a good idea to add tests in this PR?

I was thinking to split this function into two actually. first for the operator (textual representation of the operator) and the second for the value and then join. will be much more easier to read and understand. right now it looks very chaotic

Comment (IrakliJani) [reply packages/frontend/src/components/common/Filters/configs.ts:203]: I did the splitting part but test is out of the scope of the PR

**Review by owlas — APPROVED** [review]
